### ISSUE TITLE
System.IO.IOException when writing CSV File

### ISSUE THREAD

**Issue by viper3400** [viper3400/ClipBoard#10]
_Exception:_
```
Informationen über das Aufrufen von JIT-Debuggen
anstelle dieses Dialogfelds finden Sie am Ende dieser Meldung.

************** Ausnahmetext **************
System.IO.IOException: Der Vorgang ist bei einer Datei mit einem geöffneten Bereich, der einem Benutzer zugeordnet ist, nicht anwendbar.

   bei System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   bei System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   bei System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   bei System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost)
   bei System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost)
   bei System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding)
   bei System.IO.File.WriteAllLines(String path, String[] contents)
   bei ClipBoard.CsvPersistenceController.SaveToFile(String FileName, List`1 SavedItems, List`1 RecentItems)
   bei ClipBoard.MainForm.updateList()
   bei ClipBoard.MainForm.handleClipboardChanged()
   bei ClipBoard.MainForm.WndProc(Message& m)
   bei System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   bei System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   bei System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Geladene Assemblys **************
mscorlib
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1648.0 built by: NETFXREL3STAGE.
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll.
----------------------------------------
ClipBoard
    Assembly-Version: 0.0.0.10.
    Win32-Version: 0.0.0.10.
    CodeBase: file:///C:/Users/jan.graefe/PortableApps/ClipBoard/ClipBoard.exe.
----------------------------------------
Dapplo.Log
    Assembly-Version: 1.0.29.0.
    Win32-Version: 1.0.29.0.
    CodeBase: file:///C:/Users/jan.graefe/PortableApps/ClipBoard/Dapplo.Log.DLL.
----------------------------------------
System.Windows.Forms
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1586.0 built by: NETFXREL2.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll.
----------------------------------------
System
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1647.0 built by: NETFXREL3STAGE.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll.
----------------------------------------
System.Drawing
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1586.0 built by: NETFXREL2.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll.
----------------------------------------
System.Core
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1647.0 built by: NETFXREL3STAGE.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll.
----------------------------------------
FMUtils.KeyboardHook
    Assembly-Version: 1.0.140.2145.
    Win32-Version: 1.0.140.2145.
    CodeBase: file:///C:/Users/jan.graefe/PortableApps/ClipBoard/FMUtils.KeyboardHook.DLL.
----------------------------------------
System.Configuration
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1586.0 built by: NETFXREL2.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll.
----------------------------------------
System.Xml
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1586.0 built by: NETFXREL2.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll.
----------------------------------------
NLog
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.4.5361.
    CodeBase: file:///C:/Users/jan.graefe/PortableApps/ClipBoard/NLog.DLL.
----------------------------------------
System.ServiceModel
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1637.0 built by: NETFXREL3STAGE.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.ServiceModel/v4.0_4.0.0.0__b77a5c561934e089/System.ServiceModel.dll.
----------------------------------------
System.Data
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1636.0 built by: NETFXREL3STAGE.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_32/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll.
----------------------------------------
System.Runtime.Serialization
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1637.0 built by: NETFXREL3STAGE.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll.
----------------------------------------
System.Windows.Forms.resources
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1586.0 built by: NETFXREL2.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.resources/v4.0_4.0.0.0_de_b77a5c561934e089/System.Windows.Forms.resources.dll.
----------------------------------------
WindowsInput
    Assembly-Version: 1.0.4.0.
    Win32-Version: 1.0.4.0.
    CodeBase: file:///C:/Users/jan.graefe/PortableApps/ClipBoard/WindowsInput.DLL.
----------------------------------------
mscorlib.resources
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.6.1586.0 built by: NETFXREL2.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/mscorlib.resources/v4.0_4.0.0.0_de_b77a5c561934e089/mscorlib.resources.dll.
----------------------------------------

************** JIT-Debuggen **************
Um das JIT-Debuggen (Just-In-Time) zu aktivieren, muss in der
Konfigurationsdatei der Anwendung oder des Computers
(machine.config) der jitDebugging-Wert im Abschnitt system.windows.forms festgelegt werden.
Die Anwendung muss mit aktiviertem Debuggen kompiliert werden.

Zum Beispiel:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

Wenn das JIT-Debuggen aktiviert ist, werden alle nicht behandelten
Ausnahmen an den JIT-Debugger gesendet, der auf dem
Computer registriert ist, und nicht in diesem Dialogfeld behandelt.

```

**Comment by viper3400** [comment]
I created this issue in August 2017 and then thought exception message would be enough ... but don't remember how I reproduced this. In fact, there was no exception handling when writing to the csv persistence file. Now a log entry is written and a message box is shown, when there's no access to the csv file, as this is a basic precondition by design.

**Comment by viper3400** [comment]
Fixed with 5a23f68fdebe9d96836defa9e6fe660fdc9e9e37
